Lib/test/support/__init__.py

```diff
@@ -3017,20 +3017,31 @@ def is_libssl_fips_mode():
     return get_fips_mode() != 0
 
 
+_linked_to_musl = None
 def linked_to_musl():
     """
-    Test if the Python executable is linked to the musl C library.
+    Report if the Python executable is linked to the musl C library.
+
+    Return False if we don't think it is, or a version triple otherwise.
     """
+    # This is can be a relatively expensive check, so we use a cache.
+    global _linked_to_musl
+    if _linked_to_musl is not None:
+        return _linked_to_musl
+
+    # emscripten (at least as far as we're concerned) and wasi use musl,
+    # but platform doesn't know how to get the version, so set it to zero.
+    if is_emscripten or is_wasi:
+        return (_linked_to_musl := (0, 0, 0))
```

Comment:
TIL that we can also do this to save a few lines... Not sure if it's the most readable but I don't mind it.

Reply:
Yeah, I wasn't quite sure about it, but it seemed like a reasonable use of the operator.  I'm fine with being more wordy if that is considered better style.